src/wp-includes/wp-db.php

```diff
@@ -2551,20 +2551,16 @@ protected function process_fields( $table, $data, $format ) {
 				}
 			}
 
+			if ( ! function_exists( '__' ) ) {
+				wp_load_translations_early();
+			}
+
 			if ( 1 === count( $problem_fields ) ) {
-				if ( function_exists( '__' ) ) {
-					/* translators: %s Database field where the error occurred. */
-					$message = __( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contains invalid data.' );
-				} else {
-					$message = 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contains invalid data.';
-				}
+				/* translators: %s Database field where the error occurred. */
+				$message = __( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contains invalid data.' );
 			} else {
-				if ( function_exists( '__' ) ) {
-					/* translators: %s Database fields where the error occurred. */
-					$message = __( 'WordPress database error: Processing the value for the following fields failed: %s. The supplied value may be too long or contains invalid data.' );
-				} else {
-					$message = 'WordPress database error: Processing the value for the following fields failed: %s. The supplied value may be too long or contains invalid data.';
-				}
+				/* translators: %s Database fields where the error occurred. */
+				$message = __( 'WordPress database error: Processing the values for the following fields failed: %s. The supplied values may be too long or contain invalid data.' );
 			}
 
 			$this->last_error = sprintf( $message, implode( ', ', $problem_fields ) );
```

Comment:
Benchmarked using `implode()` when there's only 1 problem field vs using `reset()`. https://3v4l.org/CQgth#v7.4.25

Results: 

For 1,000,000 cycles each, `reset()` is ~2.9 ns (nanoseconds) faster than `implode()`. That's a teeny tiny difference. Memory wise, using `reset()` for a single field means `$message` variable will not be created in PHP's symtable and linked to the value memory. Readability is likely better as it is now. But given that `reset()` is slightly more performant (micro-optimization), I'll go ahead and restore that code.

Reply:
Updated in commit https://github.com/WordPress/wordpress-develop/pull/1897/commits/014e795e3e752fd0ef550e4a4483f4c4fd266486